tests/tests_deployment/test_conda_store_scaling.py

```diff
@@ -0,0 +1,245 @@
+import base64
+import json
+import logging
+import sys
+import time
+import uuid
+from unittest import TestCase
+
+import kubernetes.client
+import pytest
+import requests
+from kubernetes import client, config, dynamic
+from kubernetes.client import api_client
+from kubernetes.client.rest import ApiException
+from timeout_function_decorator import timeout
+
+from tests.tests_deployment import constants
+
+CONDA_STORE_API_ENDPOINT = "conda-store/api/v1"
+
+service_permissions = {"primary_namespace": "", "role_bindings": {"*/*": ["admin"]}}
+
+NEBARI_HOSTNAME = constants.NEBARI_HOSTNAME
+# NEBARI_HOSTNAME = "pt.quansight.dev" ## Override for local testing
+
+
+@pytest.mark.filterwarnings("ignore::urllib3.exceptions.InsecureRequestWarning")
+@pytest.mark.filterwarnings("error")
+class TestCondaStoreWorkerHPA(TestCase):
+    """
+    Creates 5 conda environments.
+    Check conda-store-worker Scale up to 5 nodes.
+    Check conda-store-worker Scale down to 0 nodes.
+    """
+
+    log = logging.getLogger()
+    logging.basicConfig(
+        format="%(asctime)s %(module)s %(levelname)s: %(message)s",
+        datefmt="%m/%d/%Y %I:%M:%S %p",
+        level=logging.INFO,
+    )
+    stream_handler = logging.StreamHandler(sys.stdout)
+    log.addHandler(stream_handler)
+
+    @staticmethod
+    def fetch_token():
+        v1 = client.CoreV1Api()
+        secret = v1.read_namespaced_secret("conda-store-secret", "dev")
+
+        token = [
+            k
+            for k in json.loads(base64.b64decode(secret.data["config.json"]))[
+                "service-tokens"
+            ].keys()
+        ][0]
+        return token
+
+    def read_namespaced_config_map(self):
+        with kubernetes.client.ApiClient(self.configuration) as _client:
+            api_instance = kubernetes.client.CoreV1Api(_client)
+        try:
+            api_response = api_instance.read_namespaced_config_map(
+                "conda-store-config", "dev"
+            )
+            return api_response
+        except ApiException as e:
+            self.log.exception(
+                "Exception when calling CoreV1Api->read_namespaced_config_map: %s\n" % e
+            )
+        finally:
+            _client.close()
+
+    def patch_namespaced_config_map(self, config_map):
```

Comment:
@viniciusdc This trick of patching the config is working when running tests against AWS but not in CI. You suggested deploy Nebari with admin token. Can override from Nebari CI config?